sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala

```diff
@@ -48,4 +52,48 @@ class DebuggingSuite extends SparkFunSuite with SharedSQLContext {
     assert(res.forall{ case (subtree, code) =>
       subtree.contains("Range") && code.contains("Object[]")})
   }
+
+  test("SPARK-28537: DebugExec cannot debug broadcast and columnar related queries") {
+    val rightDF = spark.range(10)
+    val leftDF = spark.range(10)
+    val joinedDF = leftDF.join(rightDF, leftDF("id") === rightDF("id"))
+    try {
+      val captured = new ByteArrayOutputStream()
+      Console.withOut(captured) {
+        joinedDF.debug()
+      }
+
+      val output = captured.toString()
+      assert(output.contains(
+        """== BroadcastExchange HashedRelationBroadcastMode(List(input[0, bigint, false])) ==
+        |Tuples output: 0
+        | id LongType: {}
+        |== WholeStageCodegen ==
+        |Tuples output: 10
+        | id LongType: {java.lang.Long}
+        |== Range (0, 10, step=1, splits=2) ==
+        |Tuples output: 0
+        | id LongType: {}""".stripMargin))
+    } catch {
```

Comment:
why do we need this catch? the test would fail anyway

Reply:
It's for better error message.
With `catch`, we can identify which assertion fails easily.
But if we split test cases, I think we can remove `try/catch`